### PR TITLE
Mention how to fix pixman, cairo missing library errors that occur while running tests in readthedocs

### LINKED ISSUE #13938: M1 Mac local test runs fail with "Library not loaded: '@rpath/libpixman-1.0.dylib'" dependency error
## Description

Jupyterlab test runs fail locally with "Library not loaded: '@rpath/libpixman-1.0.dylib'" error. 

- This might be specific to MacOS and/or Apple silicon machines (reproduced on 2 such machines with clean jupyterlab install in fresh conda environment)
- Might be connected to #13783, CC @fcollonval 

<details>
  <summary>Error:</summary>

  ```
  > @jupyterlab/coreutils:test

$ jest
@jupyterlab/coreutils:  FAIL  test/pageconfig.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils:  FAIL  test/url.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils:  FAIL  test/activitymonitor.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils:  FAIL  test/markdowncodeblocks.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils:  FAIL  test/path.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils:  FAIL  test/time.spec.ts
@jupyterlab/coreutils:   ● Test suite failed to run
@jupyterlab/coreutils:     dlopen(/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node, 0x0001): Library not loaded: '@rpath/libpixman-1.0.dylib'
@jupyterlab/coreutils:       Referenced from: '/Users/aieroshe/Documents/jupyterlab/node_modules/canvas/build/Release/canvas.node'
@jupyterlab/coreutils:       Reason: tried: '/usr/local/lib/libpixman-1.0.dylib' (no such file), '/usr/lib/libpixman-1.0.dylib' (no such file)
@jupyterlab/coreutils:       at Object.<anonymous> (../../node_modules/canvas/lib/bindings.js:3:18)
@jupyterlab/coreutils: Test Suites: 6 failed, 6 total
@jupyterlab/coreutils: Tests:       0 total
@jupyterlab/coreutils: Snapshots:   0 total
@jupyterlab/coreutils: Time:        0.226 s
  ```
</details>

## Reproduce

1. Attempt to build and run tests as described in ["Build and run tests"](https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#build-and-run-the-tests) section of the contributing guide, for the whole test suite or for separate package
```
jlpm run build:testutils
jlpm test
```
```
cd packages/notebook
jlpm run build:test
jlpm test --runInBand
```
2. See tests failing with an error described above

## Expected behavior

Test runs go through or fail for valid reason

## Context

Operating System and version: MacOS 12.6.3 Monterey, Apple M1 Pro chip
Browser and version: Chrome version 109.0.5414.87 (Official Build) (arm64)
JupyterLab version: 3.6.x, head-of-master (WIP 4.0.0)

### PR DESCRIPTION
**Pull request by andrii-i**
## References

Local test runs sometimes fail due to missing `pixman` and/or `cairo` (for Mac computers with Apple silicon) missing packages required by `canvas`. This commit fixes formatting of the command that fixes this problem in "Installing JupyterLab" subsection of "Contribute" section of readthedocs, adds description of the error and how to fix it to "Build and Run the Tests" subsection of the same doc. 

- Fixes #13938 

## Code changes

None, only documentation changes: [current documentation](https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#installing-jupyterlab), [updated documentation](https://jupyterlab--13956.org.readthedocs.build/en/13956/developer/contributing.html#installing-jupyterlab)

## User-facing changes

"Contribute" section of readthedocs

## Backwards-incompatible changes

None

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -444,6 +444,14 @@ appropriate package folder:
     We advice to use it as some tests are spinning a Jupyter Server that does not
     like to be executed in parallel.
 
+If you see a test run fail with ``Library not loaded: '@rpath/libpixman-1.0.dylib'`` (or different library, for example ``libcairo.2.dylib`` for Mac computers with Apple silicon) while running the
```
Comment (JasonWeill) [inline docs/source/developer/contributing.rst:447]:
```suggestion
If you see a test run fail with ``Library not loaded: '@rpath/libpixman-1.0.dylib'``
(or a different library, such as ``libcairo.2.dylib`` for Mac computers with Apple
Silicon chips) while running the
```

**Review by JasonWeill — APPROVED** [review]

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @andrii-i 

Could we change the documentation to mention at a single place the require work around to ease the maintenance and accuracy of the documentation?

For instance I would use the section `Installing Node.js` at line 263 as single source of truth (as it is a requirement to install JupyterLab and appear first when people are reading the doc) with two sections: _with conda_ and _with homebrew_. Then in the other places you could use a reference to that section.

Additionally, don't make one note stands out compare to the others. There is no reason to highlight it for all users.

**Comment by andrii-i** [comment]
@fcollonval thank you for review and feedback. Implemented comments by moving solution instructions to 'Installing Node.js' section and referring to it in other places. Also rolled back one note standing out

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -444,7 +450,16 @@ appropriate package folder:
...
+If you see a test run fail with ``Library not loaded: '@rpath/libpixman-1.0.dylib'``
+(or a different library, such as ``libcairo.2.dylib`` for Mac computers with Apple
+Silicon chips) while running the
+``jlpm test`` command above, you may be missing packages required
+by ``canvas``. Please see
+`Installing Node.js and jlpm section`_
+of this guide for instructions on how to install these packages.
+
+We use ``jest`` for all tests, so standard ``j
+est`` workflows apply.
```
Comment (fcollonval) [inline docs/source/developer/contributing.rst:462]: Small format issue - this line break is introducing a space in the final doc.

```suggestion
We use ``jest`` for all tests, so standard ``jest`` workflows apply.
```

**Review by fcollonval — COMMENTED** [review]
Thanks @andrii-i for taking into account the review.

**Review by fcollonval — APPROVED** [review]

**Comment by andrii-i** [comment]
Thanks @fcollonval

**Pull request merged**